clippy_lints/src/misc_early.rs

```diff
@@ -433,8 +433,11 @@ impl MiscEarlyLints {
         // The `line!()` macro is compiler built-in and a special case for these lints.
         let lit_snip = match snippet_opt(cx, lit.span) {
             Some(snip) => {
-                // The snip could be empty in case of expand from procedure macro
-                if snip.is_empty() || snip.contains('!') {
+                // We test if first character in snippet is a number, because the snippet could be an expansion
+                // from proc-macro like `line!()` or `#[wasm_bindgen]`.
```

Comment:
```suggestion
                // from a built-in macro like `line!()` or a proc-macro like  `#[wasm_bindgen]`.
```

Maybe also add a hint, that the `wasm_bindgen` case could be detected by checking for macros, but this is a more general way, that also filters out edge cases like `line!`